sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -904,25 +905,43 @@ class UDFSuite extends QueryTest with SharedSparkSession {
 
   test("SPARK-34663: using java.time.Period in UDF") {
     // Regular case
-    val input = Seq(java.time.Period.ofMonths(11)).toDF("p")
+    val input = Seq(java.time.Period.ofMonths(13)).toDF("p")
+      .select($"p", $"p".cast(YearMonthIntervalType(YEAR, YEAR)).as("p_y"),
+        $"p".cast(YearMonthIntervalType(MONTH, MONTH)).as("p_m"))
     val incMonth = udf((p: java.time.Period) => p.plusMonths(1))
-    val result = input.select(incMonth($"p").as("new_p"))
-    checkAnswer(result, Row(java.time.Period.ofYears(1)) :: Nil)
-    // TODO(SPARK-35777): Check all year-month interval types in UDF
-    assert(result.schema === new StructType().add("new_p", YearMonthIntervalType()))
+    val result = input.select(incMonth($"p").as("new_p"),
+      incMonth($"p_y").as("new_p_y"),
+      incMonth($"p_m").as("new_p_m"))
+    checkAnswer(result, Row(java.time.Period.ofMonths(14).normalized(),
+      java.time.Period.ofMonths(13).normalized(),
+      java.time.Period.ofMonths(14).normalized()) :: Nil)
+    assert(result.schema === new StructType()
+      .add("new_p", YearMonthIntervalType(YEAR, MONTH))
+      .add("new_p_y", YearMonthIntervalType(YEAR, MONTH))
+      .add("new_p_m", YearMonthIntervalType(YEAR, MONTH)))
```

Comment:
Just the default one:
```suggestion
      .add("new_p", YearMonthIntervalType())
      .add("new_p_y", YearMonthIntervalType())
      .add("new_p_m", YearMonthIntervalType()))
```

Reply:
Done